lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -1241,7 +1241,23 @@ func (c *ContainerProxy) Remove(vc *viccontainer.VicContainer, config *types.Con
 // once the said container has been removed. It fetches a list of volumes that are joined
 // to at least one other container, and calls the portlayer to remove this container's
 // anonymous volumes if they are dangling. Errors, if any, are only logged.
-func removeAnonContainerVols(pl *client.PortLayer, cID string, volumes map[string]struct{}) {
+func removeAnonContainerVols(pl *client.PortLayer, cID string, vc *viccontainer.VicContainer) {
+	// NOTE: these strings come in the form of <volume id>:<destination>:<volume options>
+	volumes := vc.Config.Volumes
+	// NOTE: these strings come in the form of <volume id>:<destination path>
+	namedVolumes := vc.HostConfig.Binds
+
+	// assemble white list of volume paths before processing binds. MUST be paths, as we want to move to honoring the proper metadata in the "volumes" section in the future.
+	whitelist := make(map[string]struct{}, 0)
+	for _, entry := range namedVolumes {
+		fields := strings.SplitN(entry, ":", 2)
+		if len(fields) != 2 {
+			log.Debugf("Invalid entry found in the HostConfig.Binds metadata for container %s: entry---> %s", cID, entry)
+		}
+		destPath := fields[1]
```

Comment:
assert that `len(fields) >=2` before accessing the field given we're coding for possible invalid entries.
Will be particularly relevant if we have an `""` entry in the map.

Reply:
We are using `SplitN` with a value of n==2. so in the case of an entry of `""` should only yeiled a single entry and fail the check. I will move it to n>=2. The statements are equivalent and will future proof us from the format changing 👍